airflow/auth/managers/fab/security_manager/modules/db.py

```diff
@@ -28,11 +28,12 @@
 
 from airflow import AirflowException
 from airflow.auth.managers.fab.models import Action, Permission, Resource, Role
+from airflow.www.security_manager import AirflowSecurityManager
 
 log = logging.getLogger(__name__)
 
 
-class FabAirflowSecurityManagerOverrideDb:
+class FabAirflowSecurityManagerOverrideDb(AirflowSecurityManager):
```

Comment:
yeah so one of the reasons is this
https://github.com/apache/airflow/blob/962870f891c03334227faa18e9cc00bee03c69fc/airflow/www/security_appless.py#L36
Appless cannot inherit from _only_ AirflowSecurityManager because it uses some methods in _db_.
I guess I could maybe make it inherit from FabAirflowSecurityManagerOverride ?

The right answer is probably to untangle this, see what it needs in _db_, and decouple it I suppose.

Reply:
Can you not leave it as is? Meaning `ApplessAirflowSecurityManager` inherits from `AirflowSecurityManager`?

Reply:
I was talking about the obsolete one :)

Reply:
yeah ok but that's not a durable solution since the class is obsolete. We "added" it to give some time to the users to migrate, but we shouldn't use it internally.
Moreover, the class is empty, it's basically a shortcut to `FabAirflowSecurityManagerOverride`. In that case, might as well make `ApplessAirflowSecurityManager` inherit from `FabAirflowSecurityManagerOverride`

Reply:
I am good with that too. My understanding is `ApplessAirflowSecurityManager` is used mostly for testing purposes. We might want to make it inherit from `ApplessAirflowSecurityManager` for now and later remove this dependency somehow.

What we are trying to achieve with our work in security managers is to move everything that is FAB-auth related to FAB auth manager. What's left in security manager should be global to the application regardless of the auth manager used. The end goal is to have a working (or close to working) security manager with no override. The override should be an optional piece if you want to customize the default behavior. When we are there, then I guess we could make `ApplessAirflowSecurityManager` inherit from this security manager

Reply:
`ApplessAirflowSecurityManager` is not used only for testing, it's also used in the normal course of action, for instance in this call stack:
<img width="323" alt="image" src="https://github.com/apache/airflow/assets/114772123/6bb39272-4cc6-44a5-a232-1f7c2e5cb45b">

Reply:
Yep, this is the only one I found. That's why I said "mostly" :)